backend/src/interface/mensa_parser/mod.rs

```diff
@@ -1 +1,15 @@
 //! This interface allows the reading of menus from an external source.
+
+use crate::{interface::mensa_parser::model::ParseCanteen, util::Date};
+use async_trait::async_trait;
+
+mod model;
+
+#[async_trait]
+/// Parser interface. Provides functions which return canteen structs. Canteen structs contain raw data obtained by parsing mealplans.
+pub trait MealplanParser {
+    /// Initiate a parse procedure. Returns a canteen struct containing mealplan data of the given date.
+    async fn parse(day: Date) -> Vec<ParseCanteen>;
+    /// Initiate a parse procedure. Returns a tuple containing mealplan data of the next four weeks. The tuple contains a canteen struct with the related date.
+    async fn parse_all() -> (Date, Vec<ParseCanteen>);
```

Comment:
Shouldnt there be also multiple dates returned?

Reply:
Yes, kind of.  Vec<(Date, Vec<ParseCanteen>)>